docs/concepts/recursion.rst

```diff
@@ -0,0 +1,121 @@
+
+
+.. code-block:: python
+
+    @action(reads=["parameters_to_map"], writes=["joined_results"])
+    def run_multiple_sub_apps(state: State, __context: ApplicationContext) -> State:
+        # for every parameter incoming, run a sub-application
+        results = []
+        for parameter_set in state["parameters_to_map"]:
+            subapp = (
+                ApplicationBuilder()
+                .with_actions(...)
+                .with_transitions(...)
+                .with_tracker(__context.tracker.copy())
+                .with_spawning_parent(
+                    __context.app_id,
+                    __context.sequence_id,
+                    __context.partition_key,
+                )
+                .with_entrypoint(...)
+                .with_state(...)
+                .build()
+            )
+            results.append(subapp.run(...))
+
+        joined = _join(results)
+
+        return state.update(joined_results= _join(results))
+
+Note we're simply passing through the tracker to the next app, which sets a link between them.
+The tracker knows how to record the link so its in the UI.
+
+You can also run this in parallel -- this leverages `joblib <https://joblib.readthedocs.io/en/latest/>`_'s
+Parallel and delayed functions, but you can use whatever tooling you want (multiprocessing, threading, etc...).
+
+.. code-block:: python
+
+    @action(reads=["parameters_to_map"], writes=["joined_results"])
+    def run_multiple_sub_apps(state: State, __context: ApplicationContext) -> State:
+        def run_subapp(parameter_set):
+            subapp = (
+                ApplicationBuilder()
+                .with_actions(...)
+                .with_transitions(...)
+                .with_tracker(__context.tracker.copy())
+                .with_spawning_parent(
+                    __context.app_id,
+                    __context.sequence_id,
+                    __context.partition_key,
+                )
+                .with_entrypoint(...)
+                .with_state(...)
+                .build()
+            )
+            return subapp.run(...)
+
+        results = Parallel(n_jobs=-1)(delayed(run_subapp)(parameter_set) for parameter_set in state["parameters_to_map"])
+        joined = _join(results)
+
+        return state.update(joined_results=joined)
+
+Note the following:
+
+1. You don't have to pass in partition_key, but you need to pass in the app_id and sequence_id so the app can track
+2. You currently need to run ``__context.tracker.copy()`` to ensure the tracker is passed through correctly. We will likely be relaxing this, but it could corrupt the tracking state if you forget.
+3. You may want to have more control over the application ID's -- you may want to use a stable hash of the parameters and the parent App ID
+
+For persistence, you'll want to use (3) to ensure a stable hash -- have the sub-application load its own state from the ID, resuming where it left off.
```

Comment:
we don't wire through the outer persister do we? should probably clarify what happens with persisters.

Reply:
e.g. you can instantiate different persisters, or wire through the main one, etc...